compiler/luci/pass/src/ExpandBroadcastConstPass.cpp

```diff
@@ -65,10 +64,35 @@ luci::CircleConst *create_expanded_constant(luci::CircleConst *node, luci::Circl
   auto const node_data = &node->at<loco::DataType::FLOAT32>(0);
   auto const constant_data = &constant->at<loco::DataType::FLOAT32>(0);
 
-  auto const successor_depth = successor->dim(successor->rank() - 1).value();
-  for (uint32_t d = 0; d < successor_depth; ++d)
-    std::copy(node_data, node_data + node_size, constant_data + d * node_size);
-
+  assert(successor->rank() >= 2 && successor->rank() <= 4);
```

Comment:
Please add a check that returns nullptr if this condition is false. Note that the check has to be done before node is created.

Reply:
@jinevening I applied the change to both here(for test's sake) and #14560. Please review #14560 :-D